Various Panels: Add ability to toggle legend with keyboard shortcut

alyssajoyner opened the pull request:
**What this PR does / why we need it**: This PR allows for the toggling of a graph's legend both in the configuration panel, menu dropdown and with keyboard shortcut (`p l` when hovering over the panel). 

<img width="544" alt="Screen Shot 2022-07-25 at 10 00 32 AM" src="https://user-images.githubusercontent.com/58453566/180822922-b3a205e0-8278-4b87-a8f2-ae469afc69e4.png">

<img width="778" alt="Screen Shot 2022-07-14 at 3 48 19 PM" src="https://user-images.githubusercontent.com/58453566/179093963-63bb1ea7-763e-4422-afb6-6b916173a378.png">

Fixes #49005 

**Special notes for your reviewer**: The designs for this PR are in review by the UX team.

alyssajoyner committed:
Add show/hide slider to legend panel

alyssajoyner committed:
Update branch

alyssajoyner committed:
Fix type errors

alyssajoyner committed:
Merge main

alyssajoyner committed:
Fix type errors

alyssajoyner committed:
Update toggleLegend function

alyssajoyner committed:
Add keyboard shortcut

alyssajoyner committed:
Merge main

alyssajoyner committed:
Toggle legend manually

alyssajoyner committed:
Add keyboard shortcut for toggling legend

alyssajoyner committed:
Merge main

alyssajoyner committed:
Update LegendDisplayMode to LegendVisibility

alyssajoyner committed:
Update LegendDisplayMode

alyssajoyner committed:
Update graphProps

alyssajoyner committed:
Remove unused var

alyssajoyner committed:
Update prettier

alyssajoyner committed:
Update failing tests

alyssajoyner committed:
Update snapshots

codeincarnate commented:
Just a note so it's not forgotten, assuming you stick with configuration change route going to need to make sure this has a migration for that.

alyssajoyner committed:
Update toggle icon and panel wording

alyssajoyner committed:
Update dropdown menu wording

alyssajoyner committed:
Fix prettier error

dprokop commented on public/app/core/services/keybindingSrv.ts:
```diff
@@ -283,14 +297,14 @@ export class KeybindingSrv {
-        panelInfo.panel.render();
-      }
-    });
+    // this.bindWithPanelId('p l', (panelId) => {
+    //   const panelInfo = dashboard.getPanelInfoById(panelId)!;
+
+    //   if (panelInfo.panel.legend) {
+    //     panelInfo.panel.legend.show = !panelInfo.panel.legend.show;
+    //     panelInfo.panel.render();
+    //   }
+    // });
```
```suggestion
```

dprokop commented on packages/grafana-schema/src/schema/mudball.cue:
```diff
@@ -128,7 +128,11 @@ GraphThresholdsStyleConfig: {
 
 // TODO docs
-LegendDisplayMode: "list" | "table" | "hidden" @cuetsy(kind="enum")
+LegendDisplayMode: "list" | "table" @cuetsy(kind="enum")
+
+// TODO docs
+
+LegendVisibility: "visible" | "hidden" @cuetsy(kind="enum")
```
Not sure I understand why we need this new enum. Can't we rely on the `LegendDisplayMode`? Think it would be better given that this change implicates a need for migration.

dprokop replied:
Damn, I get it - it's probably to make sure that the given legend type(list/table) stays the same when one toggles it back on.

dprokop commented on public/app/features/dashboard/components/PanelEditor/OptionsPaneCategoryDescriptor.tsx:
```diff
@@ -28,6 +28,7 @@ export class OptionsPaneCategoryDescriptor {
   addItem(item: OptionsPaneItemDescriptor) {
     item.parent = this;
     this.items.push(item);
+    // console.log(this.items)
```
```suggestion
```

dprokop commented on packages/grafana-ui/src/options/builder/legend.tsx:
```diff
@@ -11,21 +11,33 @@ export function addLegendOptions<T extends OptionsWithLegend>(
+        ],
+      },
+    })
+    .addRadio({
+      path: 'legend.showLegend',
+      name: 'Visibility',
+      category: ['Legend'],
+      description: '',
+      defaultValue: LegendVisibility.Visible,
+      settings: {
+        options: [
+          { value: LegendVisibility.Visible, label: 'Visible' },
+          { value: LegendVisibility.Hidden, label: 'Hidden' },
```
I suggest using `addBooleanSwitch` instead and placing it above the Legend Mode option.

dprokop reviewed (commented):
Left some comments - but forgot to mention that I've tested it and it works as expected, well done!

alyssajoyner committed:
Update public/app/features/dashboard/components/PanelEditor/OptionsPaneCategoryDescriptor.tsx

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

alyssajoyner committed:
Update public/app/core/services/keybindingSrv.ts

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>

codeincarnate reviewed (commented):
This looks really good. I believe it is missing a migration for the legend display mode setting

adela-almasan committed:
Tree navigation in inline editor (#52427)

adela-almasan committed:
Tree navigation in inline editor (#52427)

adela-almasan committed:
Tree navigation in inline editor (#52427)

leandro-deveikis committed:
Secret migration from Sql KV Store to Secret Plugin (#52191)

* Created PluginSecretMigrationService to be able to migrate from the secrets table from the database to the secret plugin. Added migration which takes all the secrets at the sql store and stores it in the plugin. Then deletes all the secrets from the sql
* Added secretsKVStoreSQL.GetAll() method to return all the secrets at the sql table
* Renaming kvstore_test.go as sql_test.go, adding GetAll test case. Fixing decryption of keys

leandro-deveikis committed:
Secret migration from Sql KV Store to Secret Plugin (#52191)

* Created PluginSecretMigrationService to be able to migrate from the secrets table from the database to the secret plugin. Added migration which takes all the secrets at the sql store and stores it in the plugin. Then deletes all the secrets from the sql
* Added secretsKVStoreSQL.GetAll() method to return all the secrets at the sql table
* Renaming kvstore_test.go as sql_test.go, adding GetAll test case. Fixing decryption of keys

leandro-deveikis committed:
Secret migration from Sql KV Store to Secret Plugin (#52191)

* Created PluginSecretMigrationService to be able to migrate from the secrets table from the database to the secret plugin. Added migration which takes all the secrets at the sql store and stores it in the plugin. Then deletes all the secrets from the sql
* Added secretsKVStoreSQL.GetAll() method to return all the secrets at the sql table
* Renaming kvstore_test.go as sql_test.go, adding GetAll test case. Fixing decryption of keys

iwysiu committed:
CloudWatch: fix log explorer context (#52118)

iwysiu committed:
CloudWatch: fix log explorer context (#52118)

iwysiu committed:
CloudWatch: fix log explorer context (#52118)

yuri-tceretian committed:
Alerting: Rename sender.Sender to sender.ExternalAlertmanagers (#52463)

yuri-tceretian committed:
Alerting: Rename sender.Sender to sender.ExternalAlertmanagers (#52463)

yuri-tceretian committed:
Alerting: Rename sender.Sender to sender.ExternalAlertmanagers (#52463)

torkelo committed:
FolderPage: Improve folder page to work with new nav breadcrumbs and modify Dashboard page show path based breadcrumbs"  (#52428)

* FolderPage: Progress on folder page navigation

* Dashboard to adapt breadcrumbs when opening dashboard from file storage

* add warning when topnav is not enabled

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

torkelo committed:
FolderPage: Improve folder page to work with new nav breadcrumbs and modify Dashboard page show path based breadcrumbs"  (#52428)

* FolderPage: Progress on folder page navigation

* Dashboard to adapt breadcrumbs when opening dashboard from file storage

* add warning when topnav is not enabled

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

torkelo committed:
FolderPage: Improve folder page to work with new nav breadcrumbs and modify Dashboard page show path based breadcrumbs"  (#52428)

* FolderPage: Progress on folder page navigation

* Dashboard to adapt breadcrumbs when opening dashboard from file storage

* add warning when topnav is not enabled

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

usmangt committed:
Update upgrade-grafana.md (#52406)

fixed the broken link for the release notes

usmangt committed:
Update upgrade-grafana.md (#52406)

fixed the broken link for the release notes

usmangt committed:
Update upgrade-grafana.md (#52406)

fixed the broken link for the release notes

owensmallwood committed:
Adds documentation for public dashboards under the dashboards section. (#52126)

Adds documentation for public dashboards

owensmallwood committed:
Adds documentation for public dashboards under the dashboards section. (#52126)

Adds documentation for public dashboards

owensmallwood committed:
Adds documentation for public dashboards under the dashboards section. (#52126)

Adds documentation for public dashboards

mmandrus committed:
Secrets: Add logging to track secrets migration (#52481)

* add logging to track migration

* fix lint problem

mmandrus committed:
Secrets: Add logging to track secrets migration (#52481)

* add logging to track migration

* fix lint problem

mmandrus committed:
Secrets: Add logging to track secrets migration (#52481)

* add logging to track migration

* fix lint problem

Commit:
Merge branch 'main' into time-series-legend-toggle

alyssajoyner committed:
Remove extra line from mudball.cue

alyssajoyner committed:
Remove extra line from mudball.cue

Commit:
Remove extra line from mudball.cue

Commit:
Remove extra line from mudball.cue

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=52241) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=52241) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: alyssabull<br/>:x: Alyssa Bull<br/><hr/>**Alyssa Bull** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=52241) it.</sub>

alyssajoyner committed:
Remove unnecessary space

alyssajoyner committed:
Remove unnecessary space

alyssajoyner committed:
Remove unnecessary space

alyssajoyner committed:
Remove unnecessary space

alyssajoyner committed:
Revert "Remove extra line from mudball.cue"

This reverts commit cfb4b9263b7bcef9d10fc1f39bd815ed726f6359.

alyssajoyner committed:
Revert "Remove extra line from mudball.cue"

This reverts commit cfb4b9263b7bcef9d10fc1f39bd815ed726f6359.

alyssajoyner committed:
Revert "Remove extra line from mudball.cue"

This reverts commit cfb4b9263b7bcef9d10fc1f39bd815ed726f6359.

alyssajoyner committed:
Revert "Remove extra line from mudball.cue"

This reverts commit cfb4b9263b7bcef9d10fc1f39bd815ed726f6359.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=52241) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=52241) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: alyssabull<br/>:x: Alyssa Bull<br/><hr/>**Alyssa Bull** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=52241) it.</sub>

owensmallwood committed:
Public Dashboards: Add Public Tag to Dashboard Title (#52351)

Adds Public tag to dashboard title when it has an enabled public dashboard

owensmallwood committed:
Public Dashboards: Add Public Tag to Dashboard Title (#52351)

Adds Public tag to dashboard title when it has an enabled public dashboard

owensmallwood committed:
Public Dashboards: Add Public Tag to Dashboard Title (#52351)

Adds Public tag to dashboard title when it has an enabled public dashboard

torkelo committed:
Dashboard: Fix iteration property change triggering unsaved changes warning  (#51272)

torkelo committed:
Dashboard: Fix iteration property change triggering unsaved changes warning  (#51272)

torkelo committed:
Dashboard: Fix iteration property change triggering unsaved changes warning  (#51272)

leventebalogh committed:
DataSources: refactor datasource pages to be reusable (#51874)

* refactor: move utility functions out of the redux actions

* refactor: move password handlers to the feature root

* refactor: move API related functions to an api.ts

* refactor: move components under a /components folder

* refactor: move page containers under a /pages folder and extract components

* refactor: update mocks to be easier to reuse

* refactor: move tests into a state/tests/ subfolder

* refactor: expose 'initialState' for plugins

* refactor: move generic types to the root folder of the feature

* refactor: import path fixe

* refactor: update import paths for app routes

* chore: update betterer

* refactor: fix type errors due to changed mock functions

* chore: fix mocking context_srv in tests

* refactor: udpate imports to be more concise

* fix: update failing test because of mocks

* refactor: use the new `navId` prop where we can

* fix: use UID instead ID in datasource edit links

* fix:clean up Redux state when unmounting the edit page

* refactor: use `uid` instead of `id`

* refactor: always fetch the plugin details when editing a datasource

The deleted lines could provide performance benefits, although they also make the
implementation more prone to errors. (Mostly because we are storing the information
about the currently loaded plugin in a single field, and it was not validating if it
is for the latest one).

We are planning to introduce some kind of caching, but first we would like to clean up
the underlying state a bit (plugins & datasources.

* fix: add missing dispatch() wrapper for update datasource callback

* refactor: prefer using absolute import paths

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

* fix: ESLINT import order issue

* refactor: put test files next to their files

* refactor: use implicit return types for components

* fix: remove caching from datasource fetching

I have introduced a cache to only fetch data-sources once, however as we
are missing a good logic for updating the instances in the Redux store
when they change (create, update, delete), this approach is not keeping the UI in sync.
Due to this reason I have removed the caching for now, and will reintroduce it once we have a
more robust client-side state logic.

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

leventebalogh committed:
DataSources: refactor datasource pages to be reusable (#51874)

* refactor: move utility functions out of the redux actions

* refactor: move password handlers to the feature root

* refactor: move API related functions to an api.ts

* refactor: move components under a /components folder

* refactor: move page containers under a /pages folder and extract components

* refactor: update mocks to be easier to reuse

* refactor: move tests into a state/tests/ subfolder

* refactor: expose 'initialState' for plugins

* refactor: move generic types to the root folder of the feature

* refactor: import path fixe

* refactor: update import paths for app routes

* chore: update betterer

* refactor: fix type errors due to changed mock functions

* chore: fix mocking context_srv in tests

* refactor: udpate imports to be more concise

* fix: update failing test because of mocks

* refactor: use the new `navId` prop where we can

* fix: use UID instead ID in datasource edit links

* fix:clean up Redux state when unmounting the edit page

* refactor: use `uid` instead of `id`

* refactor: always fetch the plugin details when editing a datasource

The deleted lines could provide performance benefits, although they also make the
implementation more prone to errors. (Mostly because we are storing the information
about the currently loaded plugin in a single field, and it was not validating if it
is for the latest one).

We are planning to introduce some kind of caching, but first we would like to clean up
the underlying state a bit (plugins & datasources.

* fix: add missing dispatch() wrapper for update datasource callback

* refactor: prefer using absolute import paths

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

* fix: ESLINT import order issue

* refactor: put test files next to their files

* refactor: use implicit return types for components

* fix: remove caching from datasource fetching

I have introduced a cache to only fetch data-sources once, however as we
are missing a good logic for updating the instances in the Redux store
when they change (create, update, delete), this approach is not keeping the UI in sync.
Due to this reason I have removed the caching for now, and will reintroduce it once we have a
more robust client-side state logic.

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

leventebalogh committed:
DataSources: refactor datasource pages to be reusable (#51874)

* refactor: move utility functions out of the redux actions

* refactor: move password handlers to the feature root

* refactor: move API related functions to an api.ts

* refactor: move components under a /components folder

* refactor: move page containers under a /pages folder and extract components

* refactor: update mocks to be easier to reuse

* refactor: move tests into a state/tests/ subfolder

* refactor: expose 'initialState' for plugins

* refactor: move generic types to the root folder of the feature

* refactor: import path fixe

* refactor: update import paths for app routes

* chore: update betterer

* refactor: fix type errors due to changed mock functions

* chore: fix mocking context_srv in tests

* refactor: udpate imports to be more concise

* fix: update failing test because of mocks

* refactor: use the new `navId` prop where we can

* fix: use UID instead ID in datasource edit links

* fix:clean up Redux state when unmounting the edit page

* refactor: use `uid` instead of `id`

* refactor: always fetch the plugin details when editing a datasource

The deleted lines could provide performance benefits, although they also make the
implementation more prone to errors. (Mostly because we are storing the information
about the currently loaded plugin in a single field, and it was not validating if it
is for the latest one).

We are planning to introduce some kind of caching, but first we would like to clean up
the underlying state a bit (plugins & datasources.

* fix: add missing dispatch() wrapper for update datasource callback

* refactor: prefer using absolute import paths

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

* fix: ESLINT import order issue

* refactor: put test files next to their files

* refactor: use implicit return types for components

* fix: remove caching from datasource fetching

I have introduced a cache to only fetch data-sources once, however as we
are missing a good logic for updating the instances in the Redux store
when they change (create, update, delete), this approach is not keeping the UI in sync.
Due to this reason I have removed the caching for now, and will reintroduce it once we have a
more robust client-side state logic.

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

andresmgot committed:
Azure Monitor: Add support for Metric Names variables (#52322)

* Azure Monitor: Add support for Metric Names variables

* Azure Monitor: Add support for Workspaces variables (#52323)

andresmgot committed:
Azure Monitor: Add support for Metric Names variables (#52322)

* Azure Monitor: Add support for Metric Names variables

* Azure Monitor: Add support for Workspaces variables (#52323)

IevaVasiljeva committed:
Access control: Fix org user removal for OSS users (#52473)

* allow setting team managed permissions for service accounts

* Revert "allow setting team managed permissions for service accounts"

This reverts commit 75986263976e33399c0ea94682905e0a274bc7e4.

* fix org user removal for OSS

* remove comment

IevaVasiljeva committed:
Access control: Fix org user removal for OSS users (#52473)

* allow setting team managed permissions for service accounts

* Revert "allow setting team managed permissions for service accounts"

This reverts commit 75986263976e33399c0ea94682905e0a274bc7e4.

* fix org user removal for OSS

* remove comment

evictorero committed:
GrafanaUI: Add success state to ClipboardButton (#52069)

* User Experience: apply the same pattern feedback for all copy to clipboard buttons

* add copy icon to all ClipboardButton use cases

* Change primary color for copy to clipboard in create token

* Add success button variant

* Remove copy confirmation from TableCellInspectModal because it's in the base component now

* Design tweaks to copy confirmation

 - Only change the icon to tick to avoid the button changing size
 - Change button to success green
 - Only show copy confirmation state for 2 seconds

* revert TabelCellInspectModal text button back

* revert accidental change to ShareLink

Co-authored-by: joshhunt <josh@trtr.co>

evictorero committed:
GrafanaUI: Add success state to ClipboardButton (#52069)

* User Experience: apply the same pattern feedback for all copy to clipboard buttons

* add copy icon to all ClipboardButton use cases

* Change primary color for copy to clipboard in create token

* Add success button variant

* Remove copy confirmation from TableCellInspectModal because it's in the base component now

* Design tweaks to copy confirmation

 - Only change the icon to tick to avoid the button changing size
 - Change button to success green
 - Only show copy confirmation state for 2 seconds

* revert TabelCellInspectModal text button back

* revert accidental change to ShareLink

Co-authored-by: joshhunt <josh@trtr.co>

svennergr committed:
Loki: Fix `show context` not working in some occasions (#52458)

* added missing `languageProvider.start` call

* moved `languageProvider.start`

svennergr committed:
Loki: Fix `show context` not working in some occasions (#52458)

* added missing `languageProvider.start` call

* moved `languageProvider.start`

dsotirakis committed:
CI; Update Dockerfile for `grafana/grafana-ci-deploy` container (#52516)

* Use python 3.7 for gcloud

* Update comment for reference

* Update image version

* Update gcloud components

dsotirakis committed:
CI; Update Dockerfile for `grafana/grafana-ci-deploy` container (#52516)

* Use python 3.7 for gcloud

* Update comment for reference

* Update image version

* Update gcloud components

dsotirakis committed:
CI; Update Dockerfile for `grafana/grafana-ci-deploy` container (#52516)

* Use python 3.7 for gcloud

* Update comment for reference

* Update image version

* Update gcloud components

AgnesToulet committed:
RBAC: Add RBAC for data source insights (#51496)

AgnesToulet committed:
RBAC: Add RBAC for data source insights (#51496)

dprokop commented on packages/grafana-ui/src/options/builder/legend.tsx:
```diff
@@ -11,21 +11,33 @@ export function addLegendOptions<T extends OptionsWithLegend>(
           { value: LegendDisplayMode.List, label: 'List' },
           { value: LegendDisplayMode.Table, label: 'Table' },
-          { value: LegendDisplayMode.Hidden, label: 'Hidden' },
+        ],
```
+1

nmarrs committed:
Canvas: Fix selecto box offset placement #52489

nmarrs committed:
Canvas: Fix selecto box offset placement #52489

dprokop committed:
Table: Fix type safety (#52525)

dprokop committed:
Table: Fix type safety (#52525)

sakjur committed:
Chore: Upgrade to Go version 1.17.12 (#52523)

sakjur committed:
Chore: Upgrade to Go version 1.17.12 (#52523)

sakjur committed:
Chore: Upgrade to Go version 1.17.12 (#52523)

idafurjes committed:
Chore: Move user errors to user service (#52460)

* Move user not found err to user service

* User ErrCaseInsensitive from user pkg

* User ErrUserAlreadyExists from user pkg

* User ErrLastGrafanaAdmin from user pkg

* Remove errors from model

idafurjes committed:
Chore: Move user errors to user service (#52460)

* Move user not found err to user service

* User ErrCaseInsensitive from user pkg

* User ErrUserAlreadyExists from user pkg

* User ErrLastGrafanaAdmin from user pkg

* Remove errors from model

papagian committed:
Chore: Update to the latest grafana-plugin-sdk-go and more swagger fixes (#52445)

* Fix get legacy alert response

* Swagger: Fix get folder by UID response

* Fix conflicting swagger model Alert

Reanme legacy alerting swagger model to LegacyAlert to differentiate it
from the prometheus Alert

* Bump grafana-plugin-sdk-go

* Fix get folder response

* Use go-swagger command for merging the specifications and remove merge_specs script

papagian committed:
Chore: Update to the latest grafana-plugin-sdk-go and more swagger fixes (#52445)

* Fix get legacy alert response

* Swagger: Fix get folder by UID response

* Fix conflicting swagger model Alert

Reanme legacy alerting swagger model to LegacyAlert to differentiate it
from the prometheus Alert

* Bump grafana-plugin-sdk-go

* Fix get folder response

* Use go-swagger command for merging the specifications and remove merge_specs script

asimpson committed:
CloudMonitoring: add tests around experimental UI (#52487)

* CloudMonitoring: add tests around experimental UI

I also reworked the annotation query editor UI a little bit to bring it
inline with how AzureMonitor's query editor looks.

Closes #44431

asimpson committed:
CloudMonitoring: add tests around experimental UI (#52487)

* CloudMonitoring: add tests around experimental UI

I also reworked the annotation query editor UI a little bit to bring it
inline with how AzureMonitor's query editor looks.

Closes #44431

dsotirakis committed:
CI: Refactor/Reorder `*.star` files (#52384)

* Refactor test pipelines

* Move pr, main and release starlark files under scripts/drone/event

* Move cron.star to scripts/drone/events

* Further unused code removal

* Introducing build.star

* Extract verify_drone.star

* Extract windows.star

* Extract publish.star

* Remove unused code

* Fix formatting in main.star

dsotirakis committed:
CI: Refactor/Reorder `*.star` files (#52384)

* Refactor test pipelines

* Move pr, main and release starlark files under scripts/drone/event

* Move cron.star to scripts/drone/events

* Further unused code removal

* Introducing build.star

* Extract verify_drone.star

* Extract windows.star

* Extract publish.star

* Remove unused code

* Fix formatting in main.star

dsotirakis committed:
CI: Refactor/Reorder `*.star` files (#52384)

* Refactor test pipelines

* Move pr, main and release starlark files under scripts/drone/event

* Move cron.star to scripts/drone/events

* Further unused code removal

* Introducing build.star

* Extract verify_drone.star

* Extract windows.star

* Extract publish.star

* Remove unused code

* Fix formatting in main.star

IevaVasiljeva committed:
display dashboard settings to users who are allowed to edit the dashboard (#52532)

IevaVasiljeva committed:
display dashboard settings to users who are allowed to edit the dashboard (#52532)

gabor committed:
github: add issues to logs-squad based on labels (#52533)

gabor committed:
github: add issues to logs-squad based on labels (#52533)

torkelo committed:
TopNav: Updates to create service account page and invite user  (#52480)

* Simplify logic to support both navs

* Added new file

torkelo committed:
TopNav: Updates to create service account page and invite user  (#52480)

* Simplify logic to support both navs

* Added new file

Jguer committed:
OAuth: Allow role mapping from GitHub and GitLab groups (#52407)

* OAuth: Add extract role support to github

OAuth: correct github errors

Oauth: add github tests

Oauth: Allow mapping via group memberships

Oauth: Add markdown instructions to the new mappers

fix lint

* Apply suggestions from code review

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>
Co-authored-by: Vardan Torosyan <vardants@gmail.com>

* Apply suggestions from code review

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>
Co-authored-by: Vardan Torosyan <vardants@gmail.com>

Jguer committed:
OAuth: Allow role mapping from GitHub and GitLab groups (#52407)

* OAuth: Add extract role support to github

OAuth: correct github errors

Oauth: add github tests

Oauth: Allow mapping via group memberships

Oauth: Add markdown instructions to the new mappers

fix lint

* Apply suggestions from code review

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>
Co-authored-by: Vardan Torosyan <vardants@gmail.com>

* Apply suggestions from code review

Co-authored-by: Gabriel MABILLE <gamab@users.noreply.github.com>
Co-authored-by: Vardan Torosyan <vardants@gmail.com>

JohnnyQQQQ committed:
Alerting: take datasources as external alertmanagers into consideration (#52534)

JohnnyQQQQ committed:
Alerting: take datasources as external alertmanagers into consideration (#52534)

JohnnyQQQQ committed:
Alerting: Cleanup the `alert_configuration` table on write (#51497)

JohnnyQQQQ committed:
Alerting: Cleanup the `alert_configuration` table on write (#51497)

torkelo committed:
PageLayouts: Updates dashboard section routes with navId  (#52175)

* First stab at new page layouts behind feature toggle

* Simplifying PageHeader

* Progress on a new model that can more easily support new and old page layouts

* Progress

* rename folder

* Progress

* Minor change

* fixes

* Fixing tests

* Make breadcrumbs work

* Add tests for old Page component

* Adding tests for new Page component and behavior

* fixing page header test

* Fixed test

* Moving user profile routes to navId

* PageLayouts: Updates dashboards routes with navId

* added missing navId

* AppChrome outside route

* Renaming folder

* Minor fix

* Updated

* Fixing StoragePage

* Updated

* Updating translation ids

* Updated snapshot

* update nav translation ids (yes this is confusing)

Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>
Co-authored-by: joshhunt <josh@trtr.co>

torkelo committed:
PageLayouts: Updates dashboard section routes with navId  (#52175)

* First stab at new page layouts behind feature toggle

* Simplifying PageHeader

* Progress on a new model that can more easily support new and old page layouts

* Progress

* rename folder

* Progress

* Minor change

* fixes

* Fixing tests

* Make breadcrumbs work

* Add tests for old Page component

* Adding tests for new Page component and behavior

* fixing page header test

* Fixed test

* Moving user profile routes to navId

* PageLayouts: Updates dashboards routes with navId

* added missing navId

* AppChrome outside route

* Renaming folder

* Minor fix

* Updated

* Fixing StoragePage

* Updated

* Updating translation ids

* Updated snapshot

* update nav translation ids (yes this is confusing)

Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>
Co-authored-by: joshhunt <josh@trtr.co>

gamab committed:
Docs: Fix access-control links (#52537)

gamab committed:
Docs: Fix access-control links (#52537)

alyssajoyner committed:
Merge main

alyssajoyner committed:
Merge branch 'time-series-legend-toggle' of github.com:grafana/grafana into time-series-legend-toggle

alyssajoyner committed:
Merge branch 'time-series-legend-toggle' of github.com:grafana/grafana into time-series-legend-toggle

alyssajoyner committed:
Add migration

dprokop commented on public/app/features/dashboard/state/DashboardMigrator.ts:
```diff
@@ -777,6 +777,13 @@ export class DashboardMigrator {
     }
 
+    if (oldVersion < 37) {
+      panelUpgrades.push((panel: PanelModel) => {
+        panel.options.legend.showLegend = 'visible';
+        return panel;
+      });
```
I think we also need a migration for `options.legend.displayMode === LegendDisplayMode.Hidden`. With the migration you've added, all legends will become visible regardless of the `displayMode` settings.

dprokop replied:
Actually, it's even weirder. In scenario where legend was previously hidden, the migration is setting the value to `visible` and the display mode is not set at all:

You can reproduce this using the following dashboard:

<details>
<summary>dashboard.json</summary>

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 1175,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "palette-classic"
          },
          "custom": {
            "axisColorMode": "text",
            "axisLabel": "",
            "axisPlacement": "auto",
            "barAlignment": 0,
            "drawStyle": "line",
            "fillOpacity": 0,
            "gradientMode": "none",
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineInterpolation": "linear",
            "lineWidth": 1,
            "pointSize": 5,
            "scaleDistribution": {
              "type": "linear"
            },
            "showPoints": "auto",
            "spanNulls": false,
            "stacking": {
              "group": "A",
              "mode": "none"
            },
            "thresholdsStyle": {
              "mode": "off"
            }
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 9,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "legend": {
          "calcs": [],
          "displayMode": "hidden",
          "placement": "bottom",
          "showLegend": "visible"
        },
        "tooltip": {
          "mode": "single",
          "sort": "none"
        }
      },
      "title": "Panel Title",
      "type": "timeseries"
    }
  ],
  "schemaVersion": 36,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "Hidden legend migration",
  "uid": "uWgkIagVk",
  "version": 1,
  "weekStart": ""
}
```

</details>

dprokop replied:
Ah, and please add a test for this migration :)

dprokop commented on public/app/features/dashboard/state/DashboardMigrator.ts:
```diff
@@ -777,6 +777,13 @@ export class DashboardMigrator {
 
+    if (oldVersion < 37) {
+      panelUpgrades.push((panel: PanelModel) => {
+        panel.options.legend.showLegend = 'visible';
```
```suggestion
        panel.options.legend.showLegend = LegendVisibility.Visible;
```

dprokop commented on packages/grafana-ui/src/options/builder/legend.tsx:
```diff
@@ -11,21 +11,33 @@ export function addLegendOptions<T extends OptionsWithLegend>(
+        ],
+      },
+    })
+    .addRadio({
```
Please use a boolean switch instead of radio button. Similar to how we handle boolean options in other panels. Here's Table panel for reference:

<img width="447" alt="image" src="https://user-images.githubusercontent.com/2376619/180225539-0964f3ef-c17a-4f1f-9d1c-a76b71065eea.png">

alyssabull replied:
done!

alyssajoyner committed:
Change radio to boolean switch for legend toggle

alyssajoyner committed:
Update dashboard migration

alyssajoyner committed:
Update migration

dprokop commented on public/app/features/dashboard/state/DashboardMigrator.ts:
```diff
@@ -777,6 +777,18 @@ export class DashboardMigrator {
       }
     }
 
+    if (oldVersion < 37) {
+      panelUpgrades.push((panel: PanelModel) => {
+        if (panel.options?.legend && panel.options.legend.displayMode === 'hidden') {
+          panel.options.legend.displayMode! = 'list';
+          panel.options.legend.showLegend! = false;
+        } else {
+          panel.options.legend.showLegend! = true;
+        }
+        return panel;
+      });
+    }
```
This migration is fine, but I think you haven't bumped the dashboard schema version on line 79 - hence the migration is not kicking in.

Please also add a test for this new migration.

dprokop reviewed (commented):
@alyssabull I gave it another try and it's super close :) I think when my last comments are resolved we should be good to go! Nice one:)

alyssajoyner committed:
Update migration and test

alyssajoyner committed:
merge main

alyssajoyner committed:
Merge branch 'main' into time-series-legend-toggle

alyssajoyner committed:
Add migration test

alyssajoyner committed:
Merge branch 'time-series-legend-toggle' of github.com:grafana/grafana into time-series-legend-toggle

codeincarnate commented on .drone.yml:
```diff
@@ -434,6 +434,22 @@ steps:
   image: grafana/docker-puppeteer:1.0.0
   name: test-a11y-frontend
 - commands:
+<<<<<<< HEAD
```
Uh oh! this probably shouldn't be here 😢

codeincarnate commented on public/app/features/dashboard/utils/getPanelMenu.ts:
```diff
@@ -46,6 +47,12 @@ export function getPanelMenu(
   };
 
+  const onToggleLegend = (event: React.MouseEvent<any>) => {
+    console.log(panel.options.legend)
```
Remove this

codeincarnate commented on scripts/drone/steps/lib.star:
```diff
@@ -1,7 +1,22 @@
 load('scripts/drone/vault.star', 'from_secret', 'github_token', 'pull_secret', 'drone_token', 'prerelease_bucket')
 
+<<<<<<< HEAD
```
This one too 😢

alyssajoyner committed:
Code cleanup

codeincarnate commented on scripts/drone/steps/lib.star:
```diff
@@ -2,6 +2,18 @@ load('scripts/drone/vault.star', 'from_secret', 'github_token', 'pull_secret', '
 
 grabpl_version = 'v2.9.54'
 build_image = 'grafana/build-container:1.5.8'
+=======
```
Uh oh! I'm thinking this file shouldn't have any changes

codeincarnate commented on .drone.yml:
```diff
@@ -433,6 +433,13 @@ steps:
   failure: always
   image: grafana/docker-puppeteer:1.0.0
   name: test-a11y-frontend
+- commands:
```
I'm pretty sure this shouldn't have any changes relative to main. Looks like something happened here? 🤔

alyssajoyner committed:
Restore drone.yml and lib.star to match main

codeincarnate commented on pkg/services/secrets/kvstore/plugin_mig.go:
```diff
@@ -76,7 +76,6 @@ func (s *PluginSecretMigrationService) Migrate(ctx context.Context) error {
 				return err
 			}
 		}
-		s.logger.Debug("migrated unified secrets to plugin", "number of secrets", len(allSec))
```
Also looks like this shouldn't be modified

alyssajoyner committed:
Restore plugin_mig.go to match main

codeincarnate reviewed (approved):
LGTM 🥳

towolf commented:
Thank you very much @alyssabull! Awesome work!
